### ISSUE TITLE
Better support for coroutines

### ISSUE THREAD

**Issue by okkero** [MarioAriasC/funKTionale#37]
Functions like `eitherTry` and `Try` would also be useful when catching exceptions thrown by suspending functions. The current implementations don't support this. For example, I cannot to this:

```kt
fun main(args: Array<String>) = runBlocking {
    val a = async<String>(CommonPool) {
        throw Throwable()
    }

    val result = eitherTry {
        a.await()
    }
    println(result)
}
```

This fails to compile on `a.await` because we are not in a suspending block.

I see two potential solutions:

- Add new functions (or change existing signatures) to support coroutines. E.g: `eitherTry` becomes `suspend fun <T> eitherTry(body: suspend () -> T): Either<Throwable, T>`
- Simply mark these functions `inline`.

**Comment by MarioAriasC** [comment]
Fix on 1.2

**Comment by kevinherron** [comment]
I see you marked `eitherTry` as inline in 1.2, but the reality is almost every function that accepts a lambda callback should be marked inline so that it can be used from a coroutine when the callback might suspend.
